docs/visualizing-your-sql-queries/plotting-with-seaborn.md

```diff
@@ -1,14 +1,559 @@
+
+### Example
+
+Let us travel back some decades for the purpose of this example! Suppose the local municipality wants to visually assess, using the bank data, their hypothesis that an increasing unemployment rate leads to clients opting for good-standing (status either "A" or "C"), short-term (<= 1 year) loans of lower amounts in the districts of south Moravia and north Moravia in the year 1996 by duration and status.
+
+Instead of not only creating multiple sub-scatterplots manually but also filtering the joined dataset multiple times, we can do this in one line using `relplot`! The encodings (`x` and `y`), semantics (`hue`, etc.), and facet positions (`row` and `col`) are the only arguments we have to worry about when creating the faceted figure. However, before doing so, we need to get the data in order by performing `LEFT JOIN` on all the three tables and filtering by `region`, `duration`, and `status`. We shall save this table in a CTE named
+
+```{code-cell} ipython3
+%%sql --save relplot_example
+SELECT status, duration, amount, region, unemployment_rate_96
+FROM s1.account AS a
+LEFT JOIN s1.loan AS l
+ON a.account_id = l.account_id
+LEFT JOIN s1.district AS d
+ON d.district_id = a.district_id
+WHERE region IN ('north Moravia', 'south Moravia') AND
+duration <= 24 AND status IN ('A', 'C');
+```
+
+Like before, save the CTE as a pandas `DataFrame()`:
+
+```{code-cell} ipython3
+result = %sql SELECT * FROM relplot_example
+df = result.DataFrame()
+```
+
+In this `sns.relplot`, we we assign the `col` argument to the variable `region` and the `row` argument to the variable `status` that creates a faceted figure with multiple subplots arranged across both rows and columns of the grid:
+
+```{code-cell} ipython3
+sns.relplot(
+    data=df,
+    x="unemployment_rate_96",
+    y="amount",
+    hue="duration",
+    col="region",
+    row="status",
+    height=3,
+    aspect=1.5,
+)
+```
+
+This visualization should definitely help the local municipality obtain a first glance of their hypothesis! Upon eyeballing it, we do not see any apparent correlation of unemployment rate with the loan amount, but we can see that higher duration loans lead to higher amounts. Therefore, using faceted plots, we can accelerate our EDA process and focus on important relationships in the data.
+
+However, there are still some problems with this plot. The axes labels and legend title are not descriptive enough and the plot lacks a title. We can customize `FacetGrid` figures by using Matplotlib Figure-level functions that affect all facets to reduce duplication of labels. See the functions below and consult the [`FacetGrid` documentation](https://seaborn.pydata.org/generated/seaborn.FacetGrid.html) and [Matplotlib documentation](https://matplotlib.org/stable/gallery/subplots_axes_and_figures/) to know more:
+
+```{code-cell} ipython3
+g = sns.relplot(
+    data=df,
```

Comment:
This code cell has a strange mix of code and comments. It reads like genAI assisted code. 

Please improve readability. Commenting for each line is a bit much. 

```
g = sns.relplot(
    data=df,
    x="unemployment_rate_96",
    y="amount",
    hue="duration",
    col="region",
    row="status",
    facet_kws={"margin_titles": True},
    height=4,
    aspect=1,
)
g.set(xlabel=None, ylabel=None)  # remove duplicate x and y axis labels
g.set_titles(
    row_template="Status: {row_name}", col_template="{col_name}"
)  # set row (status) and column (region) titles
g.fig.subplots_adjust(top=0.9)  # adjust the Figure position
g._legend.set_title("Duration (months)")  # hue/legend title
g.legend.set_bbox_to_anchor((1.25, 0.5))  # Shift legend to the right
g.fig.suptitle(
    "Relation between Unemployment Rate and Loan Amount by Loan", x=0.235, fontsize=12
)  # main title
g.fig.text(
    0.235,
    0.95,
    "Data subsetted on Region and Status",
    ha="right",
    va="top",
    fontsize=10,
)  # subtitle
g.fig.supylabel("Loan Amount ($)")  # y-axis label
g.fig.supxlabel("Unemployment Rate (%)")  # x-axis label
plt.show(g)
```

Reply:
This is the output of the cell after running `pkgmt format`. I will remove the comments and keep the similar structure